crates/evm/evm/src/executors/corpus.rs

```diff
@@ -321,36 +359,55 @@ impl CorpusManager {
 
         let corpus = CorpusEntry::from_tx_seq(inputs);
         let corpus_uuid = corpus.uuid;
-
+        let timestamp = corpus.timestamp;
         // Persist to disk if corpus dir is configured.
         let write_result = if self.config.corpus_gzip {
             foundry_common::fs::write_json_gzip_file(
-                corpus_dir.join(format!("{corpus_uuid}{JSON_EXTENSION}.gz")).as_path(),
+                worker_corpus
+                    .join(format!("{corpus_uuid}-{timestamp}{JSON_EXTENSION}.gz"))
+                    .as_path(),
                 &corpus.tx_seq,
             )
         } else {
             foundry_common::fs::write_json_file(
-                corpus_dir.join(format!("{corpus_uuid}{JSON_EXTENSION}")).as_path(),
+                worker_corpus.join(format!("{corpus_uuid}-{timestamp}{JSON_EXTENSION}")).as_path(),
                 &corpus.tx_seq,
             )
         };
 
         if let Err(err) = write_result {
-            debug!(target: "corpus", %err, "Failed to record call sequence {:?}", &corpus.tx_seq);
+            debug!(target: "corpus", %err, "Failed to record call sequence {:?} in worker {}", &corpus.tx_seq, self.id);
         } else {
             trace!(
                 target: "corpus",
-                "persisted {} inputs for new coverage in {corpus_uuid} corpus",
-                &corpus.tx_seq.len()
+                "persisted {} inputs for new coverage in worker {} for {corpus_uuid} corpus",
+                self.id, &corpus.tx_seq.len()
             );
         }
 
+        // Track in-memory corpus changes to update MasterWorker on sync
+        let new_index = self.in_memory_corpus.len();
+        self.new_entry_indices.push(new_index);
```

Comment:
I think this is fine, but may result in some corpus entries not getting sync'd e.g. due to a crash or ctr+c and restart. If you persist the last sync'd timestamp, it can recover from restarts by checking if there are newer entries written before a sync could occur